mesonbuild/compilers/mixins/visualstudio.py

```diff
@@ -236,10 +236,12 @@ def unix_args_to_native(cls, args: T.List[str]) -> T.List[str]:
                 continue
             if i.startswith('-LIBPATH:'):
                 i = '/LIBPATH:' + i[9:]
-            elif i.startswith('-L'):
+            elif (i.startswith('-L')
+                  and i not in {'-LN', '-LARGEADDRESSAWARE', '-LARGEADDRESSAWARE:NO', '-LTCG'}
+                  and not i.startswith(('-LINKREPRO:', '-LINKREPROTARGET:', '-LTCG:', '-LTCGOUT:'))):
                 i = '/LIBPATH:' + i[2:]
```

Comment:
`-L` is expected to point to a path, so you can check for it like:

```suggestion
            elif i.startswith('-L') and Path(i[1:]).is_dir():
                i = '/LIBPATH:' + i[1:]
```

There's a typo in the path slicing index, too, which I fixed in the suggestion.